Sming/Core/Network/SmtpClient.cpp

```diff
@@ -175,11 +175,9 @@ void SmtpClient::onReadyToSendData(TcpConnectionEvent sourceEvent)
 	case eSMTP_SendAuthResponse: {
 		// Calculate the CRAM-MD5 response
 		//     base64.b64encode("user " +hmac.new(password, base64.b64decode(challenge), hashlib.md5).hexdigest())
-		uint8_t digest[MD5_SIZE] = {0};
-		hmac_md5((const uint8_t*)authChallenge.c_str(), authChallenge.length(), (const uint8_t*)url.Password.c_str(),
-				 url.Password.length(), digest);
 
-		String token = url.User + ' ' + makeHexString(digest, MD5_SIZE);
+		auto digest = Crypto::HmacMd5::calculate(authChallenge, url.Password);
```

Comment:
Data/password are transposed. It's too easy to do this with potentially catestrophic results.
Simplest way I can think of to mitigate this problem generally is to make passwords/keys strongly typed so we have to do this:

```
Crypto::HmacMd5::calculate(Crypto::Password(password), data);
```

Reply:
I would prefer to name it `Crypto::Secret(...)` instead.

How about adding a non-static `calculate` method?
```
Crypto::HmacMd5(password).calculate(data)
```